elements/bulbs-carousel/bulbs-carousel.test.js

```diff
@@ -49,22 +64,126 @@ describe('<bulbs-video-carousel-slider>', () => {
       expect(
-        subject.children[0].querySelectorAll('bulbs-video-carousel-item')
+        slider.firstChild.querySelectorAll('bulbs-carousel-item')
       ).to.have.length(10);
     });
 
+    it('tracks previousButtons', () => {
+      expect(subject.previousButtons).to.be.instanceOf(NodeListConstructor);
+      expect(subject.previousButtons.item(0)).to.eq(previousButton);
+    });
+
+    it('tracks nextButtons', () => {
+      expect(subject.nextButtons).to.be.instanceOf(NodeListConstructor);
+      expect(subject.nextButtons.item(0)).to.eq(nextButton);
+    });
+
+    context('without a <bulbs-carousel-slider> present', () => {
+      beforeEach(() => slider.remove());
+
+      it('throws an execption', () => {
+        expect(() => subject.createdCallback()).to.throw(
+          /MUST contain a <bulbs-carousel-slider>/
+        );
+      });
+    });
+
     it('sets currentIndex to 0', () => {
       expect(subject.currentIndex).to.eql(0);
     });
+  });
+
+  describe('attachedCallback', () => {
+    context('there is an active carousel item', () => {
+      beforeEach(() => {
+        secondItem.setAttribute(
+          'href', window.location.pathname
+        );
+      });
 
-    describe('attachedCallback', () => {
-      it('slides items', () => {
-        sinon.spy(subject, 'slideItems');
+      it('pages to the active item in the carousel', () => {
         subject.attachedCallback();
-        expect(subject.slideItems).to.have.been.called;
+        expect(subject.pageToCarouselItem).to.have.been.calledWith(
+          secondItem
+        );
+      });
+    });
+
+    it('applies state', () => {
+      subject.attachedCallback();
+      expect(subject.applyState).to.have.been.called;
+    });
+  });
+
+  describe('onClick', () => {
+    beforeEach(() => {
+      sinon.spy(subject,'slideToPrevious');
+      sinon.spy(subject,'slideToNext');
+    });
+
+    context('click on previous button', () => {
+      beforeEach(() => previousButton.click());
+
+      it('slidesToPrevious', () => {
+        expect(subject.slideToPrevious).to.have.been.called;
+      });
+
+      it('applies state', () => {
+        expect(subject.applyState).to.have.been.called;
+      });
+    });
+
+    context('click on next button', () => {
+      beforeEach(() => nextButton.click());
+
+      it('slidesToNext', () => {
+        expect(subject.slideToNext).to.have.been.called;
+      });
+
+      it('applies state', () => {
+        expect(subject.applyState).to.have.been.called;
+      });
+    });
+  });
+
+  describe('getActiveCarouselItem', () => {
+    context('an item is active', () => {
+      beforeEach(() => {
+        secondItem.setAttribute(
+          'href', window.location.pathname
+        );
+      });
+
+      it('returns that item', () => {
+        expect(subject.getActiveCarouselItem()).to.eql(
+          secondItem
+        );
+      });
+    });
+
+    context('multiple items are active', () => {
```

Comment:
If for some reason the same video ended up in the recirc list twice.

Mostly paranoia here.